modules/pipeline/providers/cron/cron.service.go

```diff
@@ -285,6 +350,18 @@ func (s *provider) CronUpdate(ctx context.Context, req *pb.CronUpdateRequest) (*
 		if err := s.Daemon.AddIntoPipelineCrond(&cron); err != nil {
 			return apierrors.ErrUpdatePipelineCron.InternalError(err)
 		}
+
+		if toEdge {
```

Comment:
I think you can abstract a method for cron edge logic.
and add unit-test for this.